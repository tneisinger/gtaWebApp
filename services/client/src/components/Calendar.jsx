import React from 'react';
import BigCalendar from 'react-big-calendar';
import dates from 'react-big-calendar/lib/utils/dates';
import moment from 'moment';
import axios from 'axios';
import '../../node_modules/react-big-calendar/lib/css/react-big-calendar.css';
import '../css/calendar.css';

// select moment as the localizer for BigCalendar
BigCalendar.momentLocalizer(moment);

const CustomToolbar = (toolbar) => {
  const goToBack = () => {
    toolbar.date.setMonth(toolbar.date.getMonth() - 1);
    toolbar.onNavigate('prev');
  };

  const goToNext = () => {
    toolbar.date.setMonth(toolbar.date.getMonth() + 1);
    toolbar.onNavigate('next');
  };

  const goToCurrent = () => {
    const now = new Date();
    toolbar.date.setMonth(now.getMonth());
    toolbar.date.setYear(now.getFullYear());
    toolbar.onNavigate('current');
  };

  const label = () => {
    const date = moment(toolbar.date);
    return (
      <span>
        <b>{date.format('MMMM')}</b>
        <span> {date.format('YYYY')}</span>
      </span>
    );
  };

  return (
    <div className='rbc-toolbar'>
      <span className='rbc-btn-group'>
        <button type='button' onClick={goToBack}>prev</button>
        <button type='button' onClick={goToNext}>next</button>
      </span>
      <span className='rbc-toolbar-label'>{label()}</span>
      <span className='rbc-btn-group'>
        <button type='button' onClick={goToCurrent}>today</button>
      </span>
    </div >
  );
}

class Calendar extends React.Component {

  constructor() {
    super();
    this.state = {
      current_date: new Date(),
      events: []
    }

    this.bindScopes([
      'onNavigate'
    ]);
  }

  componentDidMount() {
    this.getEvents();
  };

  bindScopes(keys) {
    for (let key of keys) {
      this[key] = this[key].bind(this);
    }
  }

  onNavigate(date, view) {
    console.log('#### onNavigate');
    this.getVisibleDateRange(date);
  }

  getVisibleDateRange(date = this.state.current_date) {
    let result = {
      start_date: moment(dates.firstVisibleDay(date)).format('YYYY-MM-DD'),
      end_date: moment(dates.lastVisibleDay(date)).format('YYYY-MM-DD')
    }
    return result;
  }

  getEvents() {
    let date_range = this.getVisibleDateRange();
    axios.get(`${process.env.REACT_APP_FLASK_SERVICE_URL}/admin/events`, {
      params: date_range,
      headers: {
        'Authorization': `Bearer ${window.localStorage.getItem('authToken')}`
      }
    })
    .then((res) => {
      console.log(res);
    })
    .catch((err) => {
      console.log(err);
      alert('You are not authorized!');
      console.log(err.response.data);
      console.log(err.response.status);
    });
  }

  render() {
    return (
      <div className='calendar-container'>
        <BigCalendar
          defaultDate={this.state.current_date}
          onNavigate={this.onNavigate}
          events={this.state.events}
          startAccessor={'start'}
          endAccessor={'end'}
          components={{
            toolbar: CustomToolbar
          }}
        />
      </div>
    );
  }

}

export default Calendar;
