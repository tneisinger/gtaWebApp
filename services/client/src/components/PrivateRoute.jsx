import React from 'react';
import { Route } from 'react-router-dom';

const PrivateRoute = ({ component: Component, ...rest }) => {
  <Route {...rest} render={(props) => (
    props.userLoggedIn
    ? <Component {...props} />
    : { () => props.handleDenyPrivateRouteAccess(props.location) }
  )}
};

export default PrivateRoute;
