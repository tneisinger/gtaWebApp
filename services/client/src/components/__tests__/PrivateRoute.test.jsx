import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Route, Redirect } from 'react-router-dom';
import { MemoryRouter } from 'react-router';

import PrivateRoute from '../PrivateRoute';
import Calendar from '../Calendar';

jest.mock('../Calendar');
Calendar.mockImplementation(() => 'Calendar');

describe('The PrivateRoute component', () => {
  // Initialize the props that will get passed into Calendar
  let props = {
    userLoggedIn: false,
    componentProps: {
      propOne: 'first',
      propTwo: 'second',
    },
    path: "/somepath",
    component: Calendar,
  };

  let mountedPrivateRoute;

  let treeOfPrivateRoute;

  let routerHistory;

  const mounted = () => {
    if (!mountedPrivateRoute) {
      mountedPrivateRoute = mount(
        <MemoryRouter initialEntries={routerHistory}>
          <PrivateRoute {...props} />
        </MemoryRouter>,
      );
    }
    return mountedPrivateRoute;
  };

  const tree = () => {
    if (treeOfPrivateRoute) {
      treeOfPrivateRoute = renderer.create(
        <MemoryRouter initialEntries={routerHistory}>
          <PrivateRoute {...props} />
        </MemoryRouter>,
      );
    }
    return treeOfPrivateRoute;
  }

  beforeEach(() => {
    routerHistory = undefined;
    mountedPrivateRoute = undefined;
    treeOfPrivateRoute = undefined;
  });

  it('should always render a Route component', () => {
    const wrapper = shallow(<PrivateRoute {...props} />);
    expect(wrapper.find(Route).length).toBeGreaterThan(0);
  });

  describe('when userLoggedIn is false', () => {

    beforeEach(() => {
      routerHistory = ['/somepath'];
    });

    it('should redirect the client', () => {
      expect(mounted().instance().history.location.pathname).toBe("/");
    });

    it('should render properly', () => {
      expect(tree()).toMatchSnapshot();
    });
  });


  describe('when userLoggedIn is true', () => {

    beforeEach(() => {
      routerHistory = ['/somepath'];
      props.userLoggedIn = true;
    });

    it('should not redirect the client', () => {
      expect(mounted().instance().history.location.pathname).toBe("/somepath");
    });

    it('should render properly', () => {
      expect(tree()).toMatchSnapshot();
    });
  });

});
