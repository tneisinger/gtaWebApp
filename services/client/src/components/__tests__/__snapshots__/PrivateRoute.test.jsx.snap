// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PrivateRoute component when userLoggedIn is false should render properly 1`] = `undefined`;

exports[`The PrivateRoute component when userLoggedIn is true should render properly 1`] = `undefined`;
